Translatable Fields

jarednorman opened the pull request:
Some stores may not want to translate all fields, or may have additional fields that they want to translate. This allows stores to configure that themselves.

Commit:
Utilize configuration object for translatable fields

Previously, translatable fields were hard-coded in the various decorators.
With this change, they can be overridden with configuration in the client application
using similar configuration patterns to other solidus extensions.

Commit:
Add note to README explaining configuring translatable fields

Commit:
Don't use globalize friendly_id for slug unless slug is translatable

jarednorman commented:
@omnistegan I don't know if you're interested, but I would love to merge this PR, but the builds on this PR are borked... any chance you have a second to look at that?

jarednorman commented:
We're definitely going to want to merge this eventually once it gets fixed up. We're currently using it in production.

fthobe commented:
@jarednorman I saw you already used this, does it have any performance impacts when many languages / products are used?

jarednorman commented:
This PR just makes it possible for users of the extension to pick which fields are translated. The actual performance of using translated fields is unchanged.

Commit:
Remove circleCI config that uses ruby 3.0

This is no longer available for any supported solidus version

Commit:
Define ransackable_ methods on ::Translation models

After ransack v4, we need to declare these methods on the Translation
model if they are ransackable associations.

https://github.com/activeadmin/activeadmin/discussions/8033#discussioncomment-6564120

Commit:
Change tests to use query param locale instead of route

Routing translations is not directly supported by solidus globalize
and the upstream dependency doesn not support rails 7. In order to
test against newer versions of solidus, these tests are changed to use
the query param instead.

Commit:
Remove routing-filter

This gem was only utilized in the test sandbox, but users are expected
to configure locale routing themselves. Since this gem does not
support rails 7.

https://github.com/svenfuchs/routing-filter/pull/87

fthobe commented:
> This PR just makes it possible for users of the extension to pick which fields are translated. The actual performance of using translated fields is unchanged.

The question was more related to this gem overall: 
- Do you see large performance impacts with this gem? 
- Is performance driven by how much stuff I throw in here?

jarednorman commented:
If you don't watch out for N+1 queries, yes you can cause performance issues, but those can be worked around with preloading.

jarednorman commented:
Because this is "my" PR, even though none of the commits are mine, I can't approve it. 😄

jarednorman commented:
I've requested a review from @adammathys, because I'm not allowed to review this.

Commit:
Override serialize to support Rails 7.1 with Globalize 6

Globalize 6 provides an override for serialize that only suits the
deprecated signature.

This override is applied only if Rails is >= 7.1 and Globalize is < 7
Since globalize 6 does not support Rails 7.2, no handling of Globalize
6 with Rails 7.2 is needed.

The override here is from Globalize v7.0.0 and will be modified in the
next commit.

Commit:
Always pass a class_name_or_coder param to super

The Globalize 6 override that exists in the ancestor chain requires
this value to be passed from :type.

fthobe commented:
Can you also remove the tests that are not running?

harmonymjb commented:
> Can you also remove the tests that are not running?

This is the last successful test run for this PR in CI:
https://app.circleci.com/pipelines/github/solidusio-contrib/solidus_globalize/344/workflows/8755a536-a74c-4f20-a0bd-333ab8055dcb/jobs/749

No pending or skipped tests were reported. :thinking:
Can you clarify which tests you are referring to?

fthobe commented:
It's on waiting for result since yesterday. 
<img width="841" alt="Screenshot 2025-01-30 at 19 48 24" src="https://github.com/user-attachments/assets/f04a70ae-1bc7-4492-8fd4-83b3d4065de5" />

harmonymjb commented:
> It's on waiting for result since yesterday.

It's not reporting correctly on this page, but clicking into `details` shows that they passed:
![image](https://github.com/user-attachments/assets/cd5050be-967c-4303-8fcc-81b4c92b59c4)

It also reports correctly in the PR list:
![image](https://github.com/user-attachments/assets/0da94ce1-abfa-4cac-a1ac-50e7c17d83e7)

Clicking through to `View CirecleCI workflow` will show that [results are also available in CircleCI](https://app.circleci.com/pipelines/github/solidusio-contrib/solidus_globalize/344/workflows/8755a536-a74c-4f20-a0bd-333ab8055dcb?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link&utm_content=summary)

[(All runs for this PR)](https://app.circleci.com/pipelines/github/solidusio-contrib/solidus_globalize?branch=pull%2F145)

fthobe commented:
> > It's on waiting for result since yesterday.
> 
sorry, I could be more expressive. Thank you so much for your detailed reply. Yes, I saw them when going on details, but thank you very much :) 

Great work by the way and thank you.

fthobe commented:
@jarednorman this is super cool ❤️❤️❤️❤️❤️❤️

Thank you so much, it came just when we needed it.
